### PR TITLE
Remove `jsonify` and use `Response`

### PR DESCRIPTION
**Pull request by michplunkett**
## Description of Changes
Flask automatically renders the `Response` as JSON, so the `jsonify` function is redundant. Added missing `HTTPMethods` as well.

## Tests and Linting
 - [x] This branch is up-to-date with the `develop` branch.
 - [x] `pytest` passes on my local development environment.
 - [x] `pre-commit` passes on my local development environment.

### CODE REVIEW

File: OpenOversight/app/main/views.py [hunk 1]
```diff
@@ -368,7 +366,7 @@ def sitemap_officers():
 
 @main.route(
     "/officer/<int:officer_id>/assignment/new",
-    methods=[HTTPMethod.GET, HTTPMethod.POST],
+    methods=[HTTPMethod.POST],
```
Comment (michplunkett) [inline OpenOversight/app/main/views.py]: Validated by the `test_route_post_only` test.

**Comment by sea-kelp** [comment]
This seems to break some things, particularly where we are returning string where we used to return JSON errors
```bash
# Before
$ curl -X GET -H 'Accept: application/json' http://localhost:3000/asdf
{
  "error": "Not Found"
}

# After
$ curl -X GET -H 'Accept: application/json' http://localhost:3000/asdf
Not Found
```

**Comment by michplunkett** [comment]
Oof, good find. It seems to work in terms of application logic, but not so much for external calls to the API.

**Comment by michplunkett** [comment]
I may just get rid of the `Response` classes in that case and put the `jsonify` functions back in place.

**Pull request closed without merging**